lib/Common/DataStructures/BaseDictionary.h

```diff
@@ -44,20 +44,29 @@ namespace JsDiag
 
 namespace JsUtil
 {
+    struct contentStruct
+    {
+        Field(int) cs;
+    };
     class NoResizeLock
     {
     public:
-        void BeginResize() {}
-        void EndResize() {}
+#pragma prefast(suppress:__WARNING_FAILING_TO_ACQUIRE_MEDIUM_CONFIDENCE)
+        void _Acquires_lock_(cs.cs) BeginResize() {}
+#pragma prefast(suppress:__WARNING_FAILING_TO_RELEASE_MEDIUM_CONFIDENCE)
+        void _Releases_lock_(cs.cs) EndResize() {}
+    private:
+        // For prefast analysis, we need to have a somewhat similar shape for both locks
+        Field(contentStruct) cs;
```

Comment:
This is a bit of a bummer, having to waste some space for no real reason. I'm not super familiar with the semantics of SAL locking annotations; would it make sense to refer to `this` instead of the actual lock (or the fake lock here), and elsewhere treat the various lock classes as critical sections / things-that-get-locked?

Reply:
As for the memory argument, how common is this object? We tend not to worry about memory usage in objects that have very few instances.

Reply:
It's on most of our native Dictionary classes. I was also somewhat concerned that it might change the memory layout in a meaningful way, but there are some static asserts that check for that so it seems to be fine.

As for the number of native dictionaries we have, it's probably more like tens than tens of thousands, so the actual impact is probably small, but I don't have solid numbers to back that up at the moment.

I do agree that my comment here shouldn't block the PR.